### PR TITLE
Add auditing metadata in tests

### PR DESCRIPTION
**Pull request by NielsCW**

### CODE REVIEW

File: contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java [hunk 1]
```diff
@@ -138,15 +150,28 @@ void doInTransaction(ThrowingCallable callable) {
...
+                .withZone(ZoneId.of("GMT"));
+        return formatter.format(date);
+    }
+
+    public static void loginUser(String username) {
+        var user = new User(username, "password", Set.of());
+        var authentication = new UsernamePasswordAuthenticationToken(user, user.getPassword(), user.getAuthorities());
+        var context = SecurityContextHolder.getContext();
+        context.setAuthentication(authentication);
+    }
```
Comment (tgeens) [inline contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java:164]: can you explain how this is related to `@WithMockUser` on the class-level ?

Comment (NielsCW) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java:164]: `@WithMockUser` sets the `SecurityContext` to contain an authenticated `User` with the given username. This `loginUser()` overrides `SecurityContext.authentication` to contain a new authenticated `User`. So that following requests are send by the new user. This is used to test whether `lastModifiedBy` can be altered when a different user makes a PUT request for example.

Comment (tgeens) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java:164]: so the "problem" is that if you override `@WithMockUser("other")` on the test-method level, the before-each also runs with user `"other"` ?

Comment (NielsCW) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java:164]: With `@WithMockUser` alone, you can only test one user per test:
If a test is marked with `@WithMockUser("John")` and it calls a function that modifies an object in a repository with `@WithMockUser("Bob")`, `lastModifiedDate` of that object will be updated, but `lastModifiedBy` will contain `"John"`.

Comment (tgeens) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java:164]: and something like this does not work: https://docs.spring.io/spring-security/reference/servlet/test/mockmvc/authentication.html#test-mockmvc-securitycontextholder-rpp ?

**Comment by tgeens** [comment]
We need to figure out what the implications are of putting `spring-security` and `spring-boot-starter-oauth2-resource-server` on the classpath (in a separate PR)

File: contentgrid-spring-audit-logging/src/test/java/com/contentgrid/spring/audit/AuditObservationHandlerTest.java [hunk 2]
```diff
@@ -434,9 +437,13 @@ void contentDelete() throws Exception {
...
     class HandlesThrownExceptions {
 
-        private static final ServerRequestObservationContext SERVER_REQUEST_OBSERVATION_CONTEXT = new ServerRequestObservationContext(
-                new MockHttpServletRequest(),
-                new MockHttpServletResponse());
+        private static final ServerRequestObservationContext SERVER_REQUEST_OBSERVATION_CONTEXT = createContext();
+
+        private static ServerRequestObservationContext createContext() {
+            var request = new MockHttpServletRequest();
+            request.setAttribute(HandlerMapping.BEST_MATCHING_HANDLER_ATTRIBUTE, "true");
```
Comment (tgeens) [inline contentgrid-spring-audit-logging/src/test/java/com/contentgrid/spring/audit/AuditObservationHandlerTest.java:444]: why is this necessary ?

Comment (NielsCW) [reply contentgrid-spring-audit-logging/src/test/java/com/contentgrid/spring/audit/AuditObservationHandlerTest.java:444]: The tests used to work because the `handler` in `BasicAuditEventExtractor.createEventBuilder` is `null` and it was missing a null check. Since the null-check is added and the requests are crafted manually, we add this attribute to make the handler non-null and the request isn't ignored.

Comment (tgeens) [reply contentgrid-spring-audit-logging/src/test/java/com/contentgrid/spring/audit/AuditObservationHandlerTest.java:444]: isn't it fishy that all the code expect that it contains a request-handler-method and we're putting the string `"true"` in there ? :thinking:

**Review by tgeens — APPROVED** [review]
